packages/frontend/app/(auth)/register/page.tsx

```diff
@@ -0,0 +1,228 @@
+'use client'
+
+import { message, Button, Card, Col, Form, Input, Row } from 'antd'
+import React, { ReactElement, useEffect, useState } from 'react'
+import ReCAPTCHA from 'react-google-recaptcha'
+import { LeftOutlined } from '@ant-design/icons'
+import { useRouter } from 'next/navigation'
+import { userApi } from '@/app/api/userApi'
+
+export default function RegisterPage(): ReactElement {
+  const [organizationId, setOrganizationId] = useState(0)
+  const [domLoaded, setDomLoaded] = useState(false)
+  const router = useRouter()
+
+  const [registerForm] = Form.useForm()
+  const recaptchaRef = React.createRef<ReCAPTCHA>()
+
+  async function onReCAPTCHAChange(captchaCode: string | null) {
+    if (!captchaCode) return
+
+    recaptchaRef?.current?.reset()
+  }
+
+  async function createAccount() {
+    const formValues = await registerForm.validateFields()
+    const { firstName, lastName, email, password, confirmPassword, sid } =
+      formValues
+
+    if (isNaN(organizationId) || organizationId < 1) {
+      message.error('Organization not found.')
+      return
+    }
+
+    if (sid && sid.trim().length < 1) {
+      message.error('Student number must be at least 1 character')
+      return
+    }
+
+    if (password !== confirmPassword) {
+      message.error('Passwords do not match.')
+      return
+    }
+    const studentId = sid ? parseInt(sid) : null
+
+    const token = await recaptchaRef?.current?.executeAsync()
+
+    const response = await userApi.registerAccount({
+      firstName,
+      lastName,
+      email,
+      password,
+      confirmPassword,
+      sid: studentId,
+      organizationId,
+      recaptchaToken: token ?? '',
+    })
+
+    if (response.status !== 201) {
+      const data = await response.json()
+
+      message.error(data.message)
+    } else {
+      localStorage.removeItem('organizationId')
+
+      router.push('/courses')
+    }
+
+    return
+  }
+
+  useEffect(() => {
+    setDomLoaded(true)
+    setOrganizationId(parseInt(localStorage.getItem('organizationId') ?? ''))
+  }, [])
+
+  return (
+    <div>
+      {domLoaded && (
```

Comment:
avoid hydration error

Reply:
do note that hydration errors in next.js usually mean there's some illegal HTML stuff going on somewhere. 

For instance, I've encountered some with antd's Spin (with nested divs not hydrating properly), though I've fixed that by just making sure it's within <main> tags.